app/controllers/declaration/RepresentativeStatusController.scala

```diff
@@ -0,0 +1,91 @@
+import controllers.navigation.Navigator
+import forms.declaration.{RepresentativeEntity, RepresentativeStatus}
+import javax.inject.Inject
+import models.DeclarationType.DeclarationType
+import models.declaration.RepresentativeDetails
+import models.requests.JourneyRequest
+import models.{DeclarationType, ExportsDeclaration, Mode}
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
+import services.cache.ExportsCacheService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.declaration.representative_details_status
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class RepresentativeStatusController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  navigator: Navigator,
+  override val exportsCacheService: ExportsCacheService,
+  mcc: MessagesControllerComponents,
+  representativeStatusPage: representative_details_status
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport with ModelCacheable {
+
+  def displayPage(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
+    request.cacheModel.parties.representativeDetails.map(_.statusCode) match {
+      case Some(data) => Ok(representativeStatusPage(mode, navigationForm, form().fill(RepresentativeStatus(data))))
+      case _          => Ok(representativeStatusPage(mode, navigationForm, form()))
+    }
+  }
+
+  def submitForm(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    form()
+      .bindFromRequest()
+      .fold(
+        (formWithErrors: Form[RepresentativeStatus]) => Future.successful(BadRequest(representativeStatusPage(mode, navigationForm, formWithErrors))),
+        validRepresentativeDetails => updateCache(validRepresentativeDetails).map(_ => navigator.continueTo(mode, nextPage(request.declarationType)))
+      )
+  }
+
+  private def nextPage(declarationType: DeclarationType): Mode => Call =
+    declarationType match {
+      case DeclarationType.SUPPLEMENTARY =>
+        controllers.declaration.routes.DeclarationAdditionalActorsController.displayPage
+      case DeclarationType.STANDARD | DeclarationType.SIMPLIFIED | DeclarationType.OCCASIONAL | DeclarationType.CLEARANCE =>
+        controllers.declaration.routes.CarrierDetailsController.displayPage
+    }
+
+  private def navigationForm()(implicit request: JourneyRequest[AnyContent]) =
+    if (request.cacheModel.parties.representativeDetails.flatMap(_.details).isDefined) RepresentativeStatus else RepresentativeEntity
+
+  override protected def updateExportsDeclarationSync(
+    update: ExportsDeclaration => Option[ExportsDeclaration]
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext, request: JourneyRequest[_]): Future[Option[ExportsDeclaration]] =
+    super.updateExportsDeclarationSync(update)
```

Comment:
Is this method necessary? Why we can't invoke it directly?

Reply:
No idea how that method got there.  I thought it might have come from the original RepresentativeDetailsControler, but no.  It was all my own work.  I'll delete it.